### PR TITLE
chore(cdk-release): track and bump separate alpha version

### PR DESCRIPTION
**Pull request by njlynch**
As part of the project to release our alpha modules as independent modules, this
change extends our `cdk-release` tool to support tracking and bumping two
versions: the primary, stable version, and an additional, optional alpha
version.

If the local `version.vX.json` file has an `alphaVersion` property, this version
will be bumped alongside the main (stable) version. If the main version is also
a prerelease, then the alphaVersion is simply incremented (e.g., `2.0.0-alpha.0`
-> `2.0.0-alpha.1`); if the main version is stable, the alpha is incremented to
match (e.g., the `2.1.0` release will get an alpha of `2.1.0-alpha.0`). If no
`alphaVersion` is present, it is simply ignored.

This is still only part of the work to release the alpha modules. Remaining:
- Create the indepedent module CHANGELOGs, and aggregate into the main CHANGELOG
- Change the `align-versions` script to read and apply the alphaVersion if the
  module is an alpha module.
- Create the first alpha version in the v2-main branch.

*Implementation details:*

I was frustrated by the generic 'Updater' and 'UpdaterModule' pattern that I
needed to work around, so I decided to just simplify by removing the generic
capabilities that we're not using. This involved deleting a lot of code that
wasn't doing much. Lots of refactoring ensured. Similarly, there simply weren't
nearly enough tests in this module yet; I added at least a few more to improve
coverage in anticipation of the next tasks (which will require even more tests).
My apologies to the reviewer(s) for the muddled diff, but I strongly believe
these simplifications will make it easier to maintain `cdk-release` going
forward.

related #15591

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/cdk-release/lib/types.ts [hunk 1]
```diff
@@ -9,30 +9,16 @@ export interface Lifecycles {
...
-  isPrivate?: (contents: string) => string | boolean | null | undefined;
-  readVersion(contents: string): string;
-  writeVersion(contents: string, version: string): string;
-}
-
-export interface ArgUpdater {
-  filename: string;
-  type?: string;
-  updater?: UpdaterModule | string;
+export interface Version {
```
Comment (rix0rrr) [inline tools/cdk-release/lib/types.ts:16]: I would quibble with the naming here: `Version` is-a `tuple of 2 strings`, each of which is a full version string by itself... ?

Call it `Versions` at least?

Comment (rix0rrr) [reply tools/cdk-release/lib/types.ts:16]: Also perhaps `stableVersion` ?

**Review by rix0rrr — APPROVED** [review]
I have one extremely tiny nit. Will leave to you to decide.

File: tools/cdk-release/lib/types.ts (hunk 1 above)
Comment (njlynch) [reply tools/cdk-release/lib/types.ts:16]: `Versions` is fine. I _originally_ went with `stableVersion`, but wanted _slightly_ easier reading/writing to the existing versions file. I suppose it's technically more correct not to. I'll adjust. :)

**Pull request merged**